src/mrdog/config_storage.rs

```diff
@@ -1,5 +1,93 @@
-use super::ConfigItem;
+use dirs;
+use serde::{Deserialize, Serialize};
+use std::{
+    fs, io,
+    path::{self, PathBuf},
+};
 
-pub fn set_config(item: ConfigItem) {
-    todo!("Set config {item:?}")
+use super::{ConfigValue, Provider};
+
+#[derive(Debug)]
+pub enum ConfigError {
+    CreateFailed(io::Error),
+    CouldNotConvertToToml(toml::ser::Error),
+    WriteToConfigFailed(io::Error),
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+struct Config {
+    github: Option<Vec<ConfigValue>>,
+    gitlab: Option<Vec<ConfigValue>>,
```

Comment:
NIT: Now this is when the structure I drafted became retarded, I would expect here not generic `ConfigValue` but the GL token directly (or wrappign structure). It is retardness we know about and will change ruthlessly in the future.